examples/splunk2argus.py

```diff
@@ -0,0 +1,234 @@
+if not opts.quiet:
+    logging.root.setLevel(opts.quiet and logging.WARN or (opts.debug and logging.DEBUG or logging.INFO))
+
+# Conversion function for Splunk format
+def to_gmt_epoch(tsstr):
+    # tsstr is expected to be in the default Splunk format: "2015-11-01T00:00:00.000+00:00", return epoch time in ms resolution
+    return calendar.timegm(time.strptime(tsstr[:19], "%Y-%m-%dT%H:%M:%S"))*1000
+
+# Function to connect to Splunk, execute query, return results as dictionary
+def get_splunk_metrics(opts):
+
+    # Parse command-line given Splunk API endpoint
+    splunkendpoint = urlparse.urlsplit(opts.splunkapi)
+    splunk_opts = {
+        "scheme": splunkendpoint.scheme,
+        "host": splunkendpoint.hostname,
+        "port": splunkendpoint.port,
+        "username": opts.user,
+        "password": opts.password,
+    }
+
+    # Try creating a service to use for the query
+    try:
+        if not opts.quiet:
+            logging.info("Logging into Splunk service")
+        service = splunkclient.connect(**splunk_opts)
+        if not opts.quiet:
+            logging.info("Splunk login successful")
+    except:
+        logging.exception("Splunk login failed")
+        return None
+
+    # Query to execute, with runtime parameters in {}
+    splunkquery = opts.splunkquery
+    # Create and monitor job to execute the Splunk query
+    if not opts.quiet:
+        logging.info("Submitting job to Splunk..")
+    job = service.jobs.create(splunkquery)
+    if not opts.quiet:
+        logging.info("Waiting for job to be ready..")
+    while not job.is_ready():
+        if opts.verbose:
+            logging.info("Still waiting for job to be ready..")
+        time.sleep(1)
+    else:
+        if not opts.quiet:
+            logging.info("Job is ready, waiting for completion..")
+    while not job.is_done():
+        if opts.verbose:
+            logging.info("Still waiting for job to be completed..")
+        time.sleep(2)
+    else:
+        if not opts.quiet:
+            logging.info("Job is done, collecting results..")
+
+    # Assign job results, in CSV format, all records, to results
+    results = job.results(output_mode="csv", count=0)
+    # Use a CSV reader to iterate through the results, creating a list named data
+    csvr = csv.reader(results)
+    # var for column names from first row in result
+    cols = None
+    data = []
+    for row in csvr:
+        logging.info("Got row: %s", row)
+        # Assign cols from the first row, the header in the CSV
+        if not cols:
+            cols = row
+            continue
+        # Append rows to data
+        data.append(dict(zip(cols, row)))
+    if not opts.quiet:
+        logging.info("Total result count: %s", len(data))
+
+    # dictionary var to hold metrics
+    m_dict = {}
+    # for loop to populate m_dict
+    for row in data:
+
+        index = row["index"]
+
+        if not index:
+            logging.warn("Skipping row without index: %s", row)
+            continue
```

Comment:
Lines 170 to 174 seem to be leftover code? Rather, I guess you would want a generalized check inside the loop at line 181.